src/main/java/com/teamc/fms/repo/FacilityRepository.java

```diff
@@ -1,11 +1,18 @@
 package com.teamc.fms.repo;
 
 import com.teamc.fms.entity.Facility;
+import com.teamc.fms.entity.TimeSlot;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface FacilityRepository extends JpaRepository<Facility, String> {
 
+    @Query(value = "SELECT t.* FROM timeslot t WHERE t.id NOT IN (SELECT bt.timeslot_id FROM booking_timeslot bt JOIN booking b ON bt.booking_id = b.id WHERE b.booking_date BETWEEN date_trunc('day', CURRENT_DATE) AND date_trunc('day', CURRENT_DATE) + interval '7 day') AND t.facility_id = :facilityId", nativeQuery = true)
```

Comment:
We should not use native queries, as it makes problems if we try to change the underlying database, or even if we work in different dataabses
I think we should return a model containing all the time slots availabale in this date range even they are booked
So that in UI, we show both the booked and the free slots, may be like this

public class TimeSlotAvailaibilityModel {
    private Integer id;
    private Time startHour;
    private Time endHour;
   private boolean avaiable;
}